src/rewriter/mod.rs

```diff
@@ -12,13 +12,46 @@ use crate::parser::ParsingAmbiguityError;
 use crate::selectors_vm::{self, SelectorMatchingVm};
 use crate::transform_stream::*;
 use encoding_rs::Encoding;
+use mime::Mime;
+use std::borrow::Cow;
+use std::cell::Cell;
 use std::error::Error as StdError;
 use std::fmt::{self, Debug};
+use std::ops::Deref;
 use std::rc::Rc;
 use thiserror::Error;
 
 pub use self::settings::*;
 
+#[derive(Clone)]
+pub struct SharedEncoding {
```

Comment:
Move to `base` dir

Reply:
Done.  Does it make sense to also move `AsciiCompatibleEncoding` to `base/encoding.rs`?  We would export it in a different place so it is a breaking change...

Reply:
Actually, `SharedEncoding` is not exported.  It is only used internally.  Do you still want to have it in the `base` module?